src/neuroconv/nwbconverter.py

```diff
@@ -60,15 +61,48 @@ def validate_source(cls, source_data: Dict[str, dict], verbose: bool = True):
         """Validate source_data against Converter source_schema."""
         cls._validate_source_data(source_data=source_data, verbose=verbose)
 
-    def __init__(self, source_data: Dict[str, dict], verbose: bool = True):
+    def __init__(
+        self, source_data: Dict[str, dict] = None, data_interfaces: Union[List, Dict] = None, verbose: bool = True
+    ):
         """Validate source_data against source_schema and initialize all data interfaces."""
         self.verbose = verbose
-        self._validate_source_data(source_data=source_data, verbose=self.verbose)
-        self.data_interface_objects = {
-            name: data_interface(**source_data[name])
-            for name, data_interface in self.data_interface_classes.items()
-            if name in source_data
-        }
+
+        assert_msg = "Received both data_interfaces and source_data passed as arguments. Specify only one of them"
+        assert source_data is None or data_interfaces is None, assert_msg
+
+        if source_data is not None:
+            self._validate_source_data(source_data=source_data, verbose=self.verbose)
+
+            self.data_interface_objects = {
+                name: data_interface(**source_data[name])
+                for name, data_interface in self.data_interface_classes.items()
+                if name in source_data
+            }
+
+        elif data_interfaces is not None:
+            if isinstance(data_interfaces, list):
+                # Create unique names for each interface
+                counter = {interface.__class__.__name__: 0 for interface in data_interfaces}
+                total_counts = Counter([interface.__class__.__name__ for interface in data_interfaces])
+                self.data_interface_objects = dict()
+                for interface in data_interfaces:
+                    class_name = interface.__class__.__name__
+                    counter[class_name] += 1
+                    unique_signature = f"{counter[class_name]:03}" if total_counts[class_name] > 1 else ""
+                    interface_name = f"{class_name}{unique_signature}"
+                    self.data_interface_objects[interface_name] = interface
+            elif isinstance(data_interfaces, dict):
+                self.data_interface_objects = data_interfaces
+
+            else:
+                ValueError("data_interfaces musts be a list or a dict")
+
+            # Build the data interface classess on the move
+            type(self).data_interface_classes = {
+                name: interface.__class__ for name, interface in self.data_interface_objects.items()
+            }
```

Comment:
I don't want to overwrite this class attribute. Why can't we just do 

```python
self.data_interface_classes = {
                name: interface.__class__ for name, interface in self.data_interface_objects.items()
            }
```

Reply:
I thought he was opting for a different approach now: https://github.com/catalystneuro/neuroconv/pull/164#discussion_r975901112?

Reply:
@bendichter 
If you assign `data_interface_classes` as an instance attribute instead of as a class one then line 52 in the following call fails:
https://github.com/catalystneuro/neuroconv/blob/45ade589959092f3e2a9942890b1a7b1a3c70172/src/neuroconv/nwbconverter.py#L42-L56

The reason is that class methods don't have access to instance attributes and `cls.data_interface_classes.items()` is `None`.

Reply:
The other version is implemented in #169